src/main/java/io/vertx/openapi/impl/Utils.java

```diff
@@ -50,14 +49,36 @@ public static Future<JsonObject> readYamlOrJson(Vertx vertx, String path) {
         return succeededFuture(buff.toJsonObject());
       } else if ("yaml".equals(suffix) || "yml".equals(suffix)) {
         try {
-          JsonNode node = YAML_MAPPER.readTree(buff.toString());
-          return succeededFuture(new JsonObject(node.toString()));
-        } catch (JsonProcessingException e) {
+          final Yaml yaml = new Yaml(new OpenAPIYamlConstructor());
```

Comment:
yaml could be a static field, in my opinion there is no need to instantiate it every time.